port/raspberrypi/rp2xxx/src/hal/pins.zig

```diff
@@ -425,48 +418,36 @@ pub fn Pins(comptime config: GlobalConfiguration) type {
             },
         });
     }
-}
 
-pub const GlobalConfiguration = struct {
-    GPIO0: ?Pin.Configuration = null,
-    GPIO1: ?Pin.Configuration = null,
-    GPIO2: ?Pin.Configuration = null,
-    GPIO3: ?Pin.Configuration = null,
-    GPIO4: ?Pin.Configuration = null,
-    GPIO5: ?Pin.Configuration = null,
-    GPIO6: ?Pin.Configuration = null,
-    GPIO7: ?Pin.Configuration = null,
-    GPIO8: ?Pin.Configuration = null,
-    GPIO9: ?Pin.Configuration = null,
-    GPIO10: ?Pin.Configuration = null,
-    GPIO11: ?Pin.Configuration = null,
-    GPIO12: ?Pin.Configuration = null,
-    GPIO13: ?Pin.Configuration = null,
-    GPIO14: ?Pin.Configuration = null,
-    GPIO15: ?Pin.Configuration = null,
-    GPIO16: ?Pin.Configuration = null,
-    GPIO17: ?Pin.Configuration = null,
-    GPIO18: ?Pin.Configuration = null,
-    GPIO19: ?Pin.Configuration = null,
-    GPIO20: ?Pin.Configuration = null,
-    GPIO21: ?Pin.Configuration = null,
-    GPIO22: ?Pin.Configuration = null,
-    GPIO23: ?Pin.Configuration = null,
-    GPIO24: ?Pin.Configuration = null,
-    GPIO25: ?Pin.Configuration = null,
-    GPIO26: ?Pin.Configuration = null,
-    GPIO27: ?Pin.Configuration = null,
-    GPIO28: ?Pin.Configuration = null,
-    GPIO29: ?Pin.Configuration = null,
+    // Can be called at comptime or runtime
+    pub fn pins(comptime self: GlobalConfiguration) self.PinsType() {
+        var ret: self.PinsType() = undefined;
+        inline for (@typeInfo(GlobalConfiguration).Struct.fields) |field| {
+            if (@field(self, field.name)) |pin_config| {
+                if (pin_config.function == .SIO) {
+                    @field(ret, pin_config.name orelse field.name) = gpio.num(@intFromEnum(@field(Pin, field.name)));
+                } else if (pin_config.function.is_pwm()) {
+                    @field(ret, pin_config.name orelse field.name) = pwm.Pwm {
+                        .slice_number = pin_config.function.pwm_slice(),
+                        .channel = pin_config.function.pwm_channel(),
+                    };
+                }
 
-    comptime {
-        const pin_field_count = @typeInfo(Pin).Enum.fields.len;
-        const config_field_count = @typeInfo(GlobalConfiguration).Struct.fields.len;
-        if (pin_field_count != config_field_count)
-            @compileError(comptimePrint("{} {}", .{ pin_field_count, config_field_count }));
+                // TODO
```

Comment:
Please resolve a TODO, or create a FIXME with a link to a github issue where you can provide some insight.

Reply:
Created https://github.com/ZigEmbeddedGroup/microzig/issues/311 since I don't know how the ADC works and I don't want this PR to wait for me to learn.